### PR TITLE
Add initial existing conventions to docs/

### PR DESCRIPTION
**Pull request by thejcannon**

### CODE REVIEW

File: docs/Coding-Conventions.md [hunk 1]
```diff
@@ -0,0 +1,906 @@
+NOTE: This guide is a Work In Progress!
+
+[[_TOC_]]
```
Comment (thejcannon) [inline docs/Coding-Conventions.md:3]: This doesn't work in GitHub-world. 😢

**Comment by thejcannon** [comment]
I stripped anything that referenced internal links/teams/etc...

File: docs/Coding-Conventions.md (hunk 1 above)
Comment (mshafer-NI) [reply docs/Coding-Conventions.md:3]: it does on GitHub pages - which uses Kramdown

**Comment by thejcannon** [comment]
FYI @zhindes

File: docs/Coding-Conventions.md (hunk 1 above)
Comment (thejcannon) [reply docs/Coding-Conventions.md:3]: Right now, my plan is to download a VS Code Extension to do this work for me.

Eventually I think having a GitHub Action validate the TOC on PRs would be 🔥

Comment (thejcannon) [reply docs/Coding-Conventions.md:3]: We can also use a GitHub Action to just _generate_ the TOC too (at the cost of additional commits, which... 🤷‍♂️ )

**Comment by thejcannon** [comment]
I think I'd like one more approval here to make sure I didn't miss anything obvious. @neilvana @jryckman?

**Pull request merged**